### PR TITLE
Java improvements

### PR DESCRIPTION
**Pull request by jasonjoh**
## Description

Improved Java dependency processing:

- Added check to Google's Maven repository for version information. Previously the dashboard couldn't get version information on Android dependencies since they aren't listed in maven.org
- Changed service to query Maven for pom.xml dependencies. Previously this relied on GitHub's provided version information which was hit-or-miss

### Type of change
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### Testing
See new tests in MavenServiceTests.cs and RepositoriesServiceTests.cs

### CODE REVIEW

File: SamplesDashboard/SamplesDashboard/Models/MavenQuery.cs [hunk 1]
```diff
@@ -3,6 +3,27 @@
 // ------------------------------------------------------------------------------
 
 using System.Collections.Generic;
+using Newtonsoft.Json;
+
+/*
+{
```
Comment (andrueastman) [inline SamplesDashboard/SamplesDashboard/Models/MavenQuery.cs:9]: 👍

File: SamplesDashboard/SamplesDashboard/Services/MavenService.cs [hunk 2]
```diff
@@ -34,35 +38,119 @@ public MavenService(IHttpClientFactory clientFactory, IConfiguration config, IMe
...
+
+                    // Versions are listed oldest to newest, reverse so
+                    // newest is the first in the array
+                    var versions = versionNode.InnerText.Split(',').Reverse().ToArray();
+
+                    return GetLatestVersionBasedOnCurrentVersion(versions, currentVersion);
+                }
+            }
+
+            return null;
```
Comment (andrueastman) [inline SamplesDashboard/SamplesDashboard/Services/MavenService.cs:117]: nit: For consistency with other functions called
```suggestion
            return string.Empty;
```

File: SamplesDashboard/SamplesDashboard/Services/MavenService.cs [hunk 3]
```diff
@@ -34,35 +38,119 @@ public MavenService(IHttpClientFactory clientFactory, IConfiguration config, IMe
...
             if (!_cache.TryGetValue($"maven: {packageName}", out MavenQuery mavenData))
             {
                 var httpClient = _clientFactory.CreateClient();
 
                 var packageParts = packageName.Split(':');
-                var apiUrl = $"https://search.maven.org/solrsearch/select?q=g:%22{packageParts[0]}%22%20AND%20a:%22{packageParts[1]}%22&rows=1&wt=json";
+
+                // Check Google's Maven repo first for any Android packages
+                var latestVersion = await GetLatestAndroidPackageVersion(httpClient, packageParts[0], packageParts[1], currentVersion);
+                if (!string.IsNullOrEmpty(latestVersion)) return latestVersion;
```
[the remark this answers is not in the document]
Comment (jasonjoh) [reply SamplesDashboard/SamplesDashboard/Services/MavenService.cs:51]: Added in last commit.

File: SamplesDashboard/SamplesDashboard/Services/MavenService.cs [hunk 4]
```diff
@@ -34,35 +38,130 @@ public MavenService(IHttpClientFactory clientFactory, IConfiguration config, IMe
...
+                for (int i = 0; i < packageNodes.Count; i++)
+                {
+                    var node = packageNodes[i];
+                    var versionNode = node.Attributes.GetNamedItem("versions");
+
+                    // Versions are listed oldest to newest, reverse so
+                    // newest is the first in the array
+                    var versions = versionNode.InnerText.Split(',').Reverse().ToArray();
+
+                    return GetLatestVersionBasedOnCurrentVersion(versions, currentVersion);
```
[the remark this answers is not in the document]
Comment (jasonjoh) [reply SamplesDashboard/SamplesDashboard/Services/MavenService.cs]: I changed it to check if the return is empty or null and continue on to the next node in the collection

**Review by andrueastman — APPROVED** [review]

**Pull request merged**